### DISCUSSION TITLE
Clang fails to find standard library headers on Steam Deck Linux desktop mode

### DISCUSSION THREAD

**Discussion by dortamiguel** [Homebrew/discussions#4398] [Linux]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.11
ORIGIN: https://github.com/Homebrew/brew
HEAD: 931327df1f30a3a5b1aa07657c0d1070e61988f9
Last commit: 9 days ago
Core tap JSON: 10 Apr 02:31 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :2
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: octa-core 64-bit zen
Clang: 16.0.1
Git: 2.38.1 => /bin/git
Curl: 7.85.0 => /bin/curl
Kernel: Linux 5.13.0-valve36-1-neptune x86_64 GNU/Linux
OS: SteamOS Holo (Holo)
Host glibc: 2.36
/usr/bin/gcc: N/A
/usr/bin/ruby: N/A
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I'm running the archlinux based steam deck distro, since the file system is read-only hombrew is the perfect package manager, since I can just run it from my home. The thin is that I installed clang using this command `brew install llvm` and that gave me clang++.

```
steamdeck% clang++ -v
Homebrew clang version 16.0.1
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/linuxbrew/.linuxbrew/bin
```

The thing is that when I try to run it to compile a very simple program like this
```c++
#include <iostream>
#include <string>

int main() {
    std::string str = "C++ String";
    std::cout << str << "\n";
}
```

I get this error
```sh
steamdeck% clang++ -std=c++20 main.cpp       
main.cpp:1:10: fatal error: 'iostream' file not found
#include <iostream>
         ^~~~~~~~~~
1 error generated.
```
Is there any other package I'm missing that I should install?

**Comment by carlocab** [comment]
It needs an existing host toolchain to work. You need to do the equivalent of `apt install build-essential` for your distro.

**Reply by dortamiguel** [reply]
I think I can do that by unlocking the file system of the steam deck, but it will defeat the purpose I have for using homebrew I think. Do you think it might be possible to build this extra dependencies without having to need root?

**Reply by carlocab** [reply]
It might work if you can find a way to install your system headers somewhere and add it to your include path. For example, if you install them to `$HOME/system/include` then you need to add `-isystem${HOME}/include`.
